### PR TITLE
refactor: send request to create a feed if filters exist

### PR DESCRIPTION
**Pull request by sshanzel**
Only when this PR is merged I will need to update the fetching of local storage: https://github.com/dailydotdev/apps/pull/791/files

### CODE REVIEW

File: packages/shared/src/components/profile/ProfileForm.tsx [hunk 1]
```diff
@@ -123,6 +131,28 @@ export default function ProfileForm({
       const filledFields = Object.keys(data).filter(
         (key) => data[key] !== undefined && data[key] !== null,
       );
+      if (shouldShowMyFeed) {
```
Comment (rebelchris) [inline packages/shared/src/components/profile/ProfileForm.tsx:134]: Just a note here, I think we can remove this check.
There could be an edge case where the users AB test might have been removed/changed.

I think if they have local feed settings, we should just always apply them anyway.
What do you think?

Comment (sshanzel) [reply packages/shared/src/components/profile/ProfileForm.tsx:134]: Very sharp thinking. Let me remove this one.

**Review by rebelchris — APPROVED** [review]
Nice clean, simple and it does the trick 💖

**Review by idoshamun — CHANGES_REQUESTED** [review]
Let's utilize GraphQL power.
Also I'm missing relevant tests

File: packages/shared/src/components/profile/ProfileForm.tsx [hunk 2]
```diff
@@ -123,6 +127,26 @@ export default function ProfileForm({
...
+        : Promise.resolve();
+      const onUpdateAdvancedSettings = advancedSettings?.length
+        ? updateAdvancedSettings({ advancedSettings })
+        : Promise.resolve();
+      await Promise.all([
+        onFollowTags,
+        onBlockTags,
+        onUnfollowSource,
+        onUpdateAdvancedSettings,
+      ]);
```
[the remark this answers is not in the document]
Comment (sshanzel) [reply packages/shared/src/components/profile/ProfileForm.tsx:149]: Oh, I totally agree.

**Review by rebelchris — APPROVED** [review]
Wow amazing how much we could refactor to one graphQL query.

File: packages/shared/src/components/profile/ProfileForm.tsx [hunk 3]
```diff
@@ -123,6 +130,14 @@ export default function ProfileForm({
       const filledFields = Object.keys(data).filter(
         (key) => data[key] !== undefined && data[key] !== null,
       );
+
+      const feedSettings = getLocalFeedSettings(true);
+
+      if (feedSettings) {
+        await updateFeedFilters(feedSettings);
```
Comment (rebelchris) [inline packages/shared/src/components/profile/ProfileForm.tsx:137]: As mentioned I think we should set the update filter here if the user had any filters.

**Review by rebelchris — CHANGES_REQUESTED** [review]
A new remark in combination with the discussion in the other branch.
The user already has filters, so it's fair to say we should update the alert for them.

As this would have happened when the filtered while being logged in.

**Comment by sshanzel** [comment]
So cool to see we have batched 3 requests into 1.

@idoshamun should now be ready for your sharp eyes 🦅 

@rebelchris I have now included the mutation for filters as we have confirmed there would be no issue in this part.

File: packages/shared/src/graphql/feedSettings.ts [hunk 4]
```diff
@@ -98,6 +98,23 @@ export const FEED_SETTINGS_QUERY = gql`
...
+  ) {
+    addFiltersToFeed(filters: $filters) {
+      id
+    }
+    updateFeedAdvancedSettings(settings: $settings) {
+      id
+    }
+    updateUserAlerts(data: { filter: false }) {
+      filter
+    }
```
Comment (rebelchris) [inline packages/shared/src/graphql/feedSettings.ts:114]: Wow nice!!

Is it still correct to do this manually in the sidebar? 
Or should we maybe consider later on refactoring to something similar?

File: packages/shared/src/components/profile/ProfileForm.spec.tsx [hunk 5]
```diff
@@ -37,24 +45,54 @@ const defaultUser = {
...
+  blockedTags: ['javascript'],
+  includeTags: ['react'],
+  excludeSources: [{ id: 'test', name: 'Source', image: 'a.b.c' }],
+};
+
+const createUpdateFeedFiltersMock = (
+  mock: ReturnType<typeof jest.fn>,
+  { advancedSettings, ...rest }: FeedSettings = defaultFeedSettings,
+) => {
+  const variables = { filters: rest, settings: advancedSettings };
```
Comment (rebelchris) [inline packages/shared/src/components/profile/ProfileForm.spec.tsx:59]: Perhaps we should include the alerts setting in this test as well?
What do you think?

**Review by rebelchris — CHANGES_REQUESTED** [review]
Wow Lee!

Impressive work and so happy to see with could just enable this in GraphQL 🌟 
One little note on testing for the alert, not sure what you think on that point?

File: packages/shared/src/graphql/feedSettings.ts (hunk 4 above)
Comment (sshanzel) [reply packages/shared/src/graphql/feedSettings.ts:114]: I am planning to refactor to apply it on users having the experiment disabled as we still need support the case outside experiment.

File: packages/shared/src/components/profile/ProfileForm.spec.tsx (hunk 5 above)
Comment (sshanzel) [reply packages/shared/src/components/profile/ProfileForm.spec.tsx:59]: I tried but it fails, I think it is due to the fact we are mocking the parameters and the value was strongly typed.

**Review by rebelchris — APPROVED** [review]
Nice, well done on this Lee!

**Comment by sshanzel** [comment]
Thanks for the awesome reviews @rebelchris! 🚢

**Review by idoshamun — APPROVED** [review]
Sweet! Let's merge :)
